changelog.d/8101.removal

```diff
@@ -0,0 +1 @@
+Disallow `:` characters from being included in `client_secret` parameters across all related endpoints. This is a breaking change.
```

Comment:
Not really sure if `.removal` is the best thing for this or not :thinking:

Reply:
I'd probably say a bugfix? I don't think this is very friendly to users who might read the release notes though. Maybe:

> Synapse now correctly enforces the valid characters in the `client_secret` parameter used in various endpoints.

I don't think much more is needed since you added a note to changes already.